compiler/rustc_borrowck/src/type_check/mod.rs

```diff
@@ -1346,7 +1346,12 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
                     Some(RETURN_PLACE) => {
                         if let BorrowCheckContext {
                             universal_regions:
-                                UniversalRegions { defining_ty: DefiningTy::Const(def_id, _), .. },
+                                UniversalRegions {
+                                    defining_ty:
+                                        DefiningTy::Const(def_id, _)
```

Comment:
can we extract this into a method, so that we write

`defining_ty.is_any_const()`

?

Reply:
There is `is_const` that does this, but this one needs to match def_id. I'll split it into a `is_const` call followed by a `def_id` call.